lib/services/internalEngine/bootstrap.js

```diff
@@ -22,310 +22,171 @@
-              // createOrReplace instead of create to avoid mess on concurency
-              // TODO fix with a lock instead
-              promises.push(
-                this.kuzzle.internalEngine.createOrReplace(
-                  'validations',
-                  `${indexName}#${collectionName}`,
-                  {
-                    index: indexName,
-                    collection: collectionName,
-                    validation: validationConfiguration[indexName][collectionName]
-                  }
-                ));
-            });
-          });
-
-          return Bluebird.all(promises);
+    return new Promise((resolve, reject) => {
+      setTimeout(async () => {
+        try {
+          await this.checkTimeout(attempts + 1);
+          resolve();
+        } catch (error) {
+          reject(error);
         }
-
-        return null;
-      })
-      .then(() => this.kuzzle.indexCache.add(this.db.index, 'validations'));
-  }
-
-  * _getJWTSecretGen () {
-    if (this.kuzzle.config.security.jwt.secret !== null) {
-      return this.kuzzle.config.security.jwt.secret;
-    }
-
-    // try to create first to avoid collisions
-    try {
-      const seed = crypto.randomBytes(512).toString('hex');
-      yield this.db.create('config', jwtSecretId, {seed});
-      this.kuzzle.config.security.jwt.secret = seed;
-    }
-    catch(e) {
-      const response = yield this.db.get('config', jwtSecretId);
-      this.kuzzle.config.security.jwt.secret = response._source.seed;
-    }
+      }, delay);
+    });
   }
 
-  adminExists () {
-    return this.db.search(
-      'users',
-      {query:
-        {terms:
-          {profileIds: ['admin']}}},
-      {from: 0, size: 0})
-      .then(response => (response.total > 0));
+  /**
+   * Delete the bootstrap lock
+   *
+   * @returns {Promise}
+   */
+  unlock () {
+    return this.storage.delete('config', LOCK_ID);
   }
 
   /**
-   * Checks if Kuzzle is already being bootstraped.
-   * If not set a lock.
-   * Returns true if the bootstrap is locked, false otherwise.
+   * Try to get the bootstrap lock to initiate bootstrap sequence.
+   * Lock older than 30 seconds are not considered valid and will be taken over.
    *
-   * @generator
-   * @private
-   * @returns {boolean}
+   * @returns {Promise<boolean>} - True if lock was acquired, false otherwise
    */
-  * _lockGen () {
+  async getLock () {
     let lock;
 
     try {
-      yield this.db.create('config', lockId, {timestamp: Date.now()});
-      return false;
-    }
-    catch (e) {
-      // lock exists - try to get it and check if it is old enough to erase it
-    }
-
-    lock = yield this.db.get('config', lockId);
-    if (lock._source.timestamp < Date.now() - 30000) {
-      yield this.db.createOrReplace('config', lockId, {timestamp: Date.now()});
-      return false;
-    }
+      lock = await this.storage.get('config', LOCK_ID);
+    } catch (error) {
+      // TODO check error code instead
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)